pkg/asset/agent/mirror/registriesconf.go

```diff
@@ -198,31 +239,33 @@ func (i *RegistriesConf) Load(f asset.FileFetcher) (bool, error) {
 
 func (i *RegistriesConf) finish() error {
 
-	config, err := extractLocationMirrorDataFromRegistries(i.File.Data)
-	if err != nil {
-		return errors.Wrap(err, fmt.Sprintf("failed to parse mirrors in %s", RegistriesConfFilename))
+	if string(i.File.Data) != defaultRegistriesConf {
+		i.validateReleaseImageIsSameInRegistriesConf()
 	}
 
-	i.MirrorConfig = config
-
 	return nil
 }
 
-// From assisted-service pkg/mirrorregistries/generator.go
-func extractLocationMirrorDataFromRegistries(registriesConfToml []byte) ([]RegistriesConfig, error) {
-	registries := sysregistriesv2.V2RegistriesConf{}
-	err := toml.Unmarshal(registriesConfToml, &registries)
-	if err != nil {
-		return nil, err
-	}
+func (i *RegistriesConf) validateReleaseImageIsSameInRegistriesConf() {
 
-	registriesConfList := make([]RegistriesConfig, len(registries.Registries))
-	for i, reg := range registries.Registries {
-		registriesConfList[i] = RegistriesConfig{
-			Location: reg.Location,
-			Mirror:   reg.Mirrors[0].Location,
+	var found bool
+
+	for _, registry := range i.Config.Registries {
+		source := registry.Endpoint.Location
+		if !i.Generated && source == "" {
+			logrus.Warnf(fmt.Sprintf("%s should have an entry matching the releaseImage %s. Found \"%s\"", RegistriesConfFilename, i.ReleaseImagePath, source))
```

Comment:
As this check doesn't compare the ReleaseImage but just the key the warning should be something like "location key not found in %s

Reply:
We should only warn if the release image path isn't found in _any_ registry entry. By checking it inside the loop here, we require it to be found in _every_ registry entry to avoid triggering the warning.